### ISSUE TITLE
support google vertexAI OpenAI API

### ISSUE THREAD

**Issue by pchalasani** [langroid/langroid#566]
https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/call-vertex-using-openai-library#client-setup

**Comment by AbhishekRP2002** [comment]
hi @pchalasani  if this is still open i would love to take this up

**Comment by pchalasani** [comment]
@AbhishekRP2002 we recently added support for using Gemini models via a single GEMINI_API_KEY but maybe the vertexAI support offers more, I haven't had time to dig in. If you think this provides useful additional functionality, happy to take a PR, thanks!

**Comment by pchalasani** [comment]
BTW, if you still want to add the metaphor -> exa change or the weaviate support, those are still open.

**Comment by AbhishekRP2002** [comment]
yes i will actually start with exa change and raise a PR by this weekend.

**Comment by pchalasani** [comment]
We can have users specify `chat_model=vertexai/google/gemini-2.0-flash-001` for example, and it will activate the `vertexai` path in `openai_gpt.py`.
